apply/v2/applydriver/local.go

```diff
@@ -165,22 +166,30 @@ func (c *Applier) upgradeCluster(mj, nj []string) error {
 		return err
 	}
 	// fetch form exec machine
-	desiredMetadata, err := runtime.LoadMetadata(filepath.Join(common.DefaultMountCloudImageDir(c.ClusterDesired.Name),
-		common.DefaultMetadataName))
+	rt, err := runtime.NewDefaultRuntime(c.ClusterDesired, c.ClusterDesired.GetAnnotationsByKey(common.ClusterfileName))
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to init runtime, %v", err)
+	}
+	KubeadmConfig := &runtime.KubeadmConfig{}
```

Comment:
Apply module should not care about kubeadmConfig, what if runtime is k3s? You can add a function in Interface like GetClusterMetadata.